File: api/src/main/java/net/blueshell/api/common/enums/FileType.java
package net.blueshell.api.common.enums;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;

@Schema(enumAsRef = true)
@Getter
public enum FileType {
    DOCUMENT("documents"),
    PROFILE_PICTURE("profile-pictures"),
    EVENT_BANNER("event-banners"),
    EVENT_PICTURE("event-pictures"),
    SPONSOR_PICTURE("sponsor-pictures");

    private final String directory;

    FileType(String directory) {
        this.directory = directory;
    }
}

File: api/src/main/kotlin/net/blueshell/api/common/enums/FileType.kt
package net.blueshell.api.common.enums

import io.swagger.v3.oas.annotations.media.Schema
import lombok.Getter

@Schema(enumAsRef = true)
@Getter
enum class FileType(directory: String) {
    DOCUMENT("documents"),
    PROFILE_PICTURE("profile-pictures"),
    EVENT_BANNER("event-banners"),
    EVENT_PICTURE("event-pictures"),
    SPONSOR_PICTURE("sponsor-pictures");

    private val directory: String?

    init {
        this.directory = directory
    }
}
